Feacture/pending requirements

lrojas870 opened the pull request:
En este pull request se agregaron las siguiente tareas segun el board del proyecto:

- Agregar una primera pantalla como splash con la imagen adjunta
- Agregar un loading a la pantalla home
- Crear date picker
- Integrar date picker con funcionalidad que busca la informacion
- Agregar validación en la fecha seleccionada en el date picker

lrojas870 committed:
Feat:
- Add image splash.png to SplashActivity.kt
- leave SplashActivity.kt as activity launcher
- rename MainActivity.kt to HomeActivity.kt

lrojas870 committed:
Feat:
- add loading to HomeActivity.kt

lrojas870 committed:
Feat:
- create class DatePickerFragment.kt
- add datePickerDialog to HomeActivity.kt
- Associate datePickerDialog to function getData

lrojas870 committed:
Feat:
- extract functions from HomeActivity.kt that do not
correspond to the view.

lrojas870 committed:
Feat:
- add validation on the selected date in the date picker

iMoya917 commented on app/src/main/java/com/androidavanzado/covid19stadistics/util/MonthToNameMonth.kt:
```diff
@@ -0,0 +1,25 @@
+
+    operator fun invoke(month: String) : String {
+
+        val nameMonth = when (month) {
```
You could use the date formats and not have a when to know the name of the month.

lrojas870 replied:
done

manuel28g-carsales commented on app/src/main/java/com/androidavanzado/covid19stadistics/storage/remote/RemoteDataSource.kt:
```diff
@@ -13,17 +13,15 @@ class RemoteDataSource(private val apiClient: ApiClient) : DataSource {
                 }
                 else{
-                    Result.Failure("Sin datos")
+                    Result.Failure("No se encontraron datos para la fecha seleccionada.")
```
Try don't use a hardcoded string, instead use the resource file

lrojas870 replied:
done

manuel28g-carsales commented on app/src/main/java/com/androidavanzado/covid19stadistics/storage/remote/RemoteDataSource.kt:
```diff
@@ -13,17 +13,15 @@ class RemoteDataSource(private val apiClient: ApiClient) : DataSource {
                 }
             }?:run{
-                return Result.Failure("Ocurrió un error")
+                return Result.Failure("Ocurrió un error, tratanto de obtener la información.")
```
Same here, use the res/values/strings file for this cases

lrojas870 replied:
done

manuel28g-carsales commented on app/src/main/java/com/androidavanzado/covid19stadistics/ui/DatePickerFragment.kt:
```diff
@@ -0,0 +1,28 @@
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+
+        val c = Calendar.getInstance()
```
change this variable name to another one more verbose like "calendar"

lrojas870 replied:
done

manuel28g-carsales commented on app/src/main/java/com/androidavanzado/covid19stadistics/util/DateTextFormat.kt:
```diff
@@ -0,0 +1,22 @@
+
+    @SuppressLint("SimpleDateFormat")
+    operator fun invoke(date: Date): String {
+        val dayFormat: DateFormat = SimpleDateFormat("dd")
```
This assignment of SimpleDateFormat should be in a global scope to be created once and not each time when call invoke method

lrojas870 replied:
done

manuel28g-carsales commented on app/src/main/java/com/androidavanzado/covid19stadistics/ui/HomeActivity.kt:
```diff
@@ -0,0 +1,135 @@
+            callGetData(dateString)
+        }
+        else{
+            Toast.makeText(this, "No puede seleccionar una fecha igual o mayor a hoy.", Toast.LENGTH_SHORT).show()
```
Move all the string of this file to the res/values/string source

lrojas870 replied:
done

lrojas870 committed:
Feat:
- move hardcoded string to strings.xml file
- user date format to get month name
- delete MonthToNameMonth.kt file

lrojas870 committed:
Feat:
- change toast for dialog
- make global variables the dateFormat in DateTextFormat.kt
